### PR TITLE
SA-419 Upgrade the Initializr to Java 17

### PR DESCRIPTION
**Pull request by atomfinger**
There is also a minor restructuring of the readme file so that it'll be prettier in both the initializr and when generated.

As for SonarQube:

1. SonarQube still lacks support for Java 16 and gets stuff like `Record` wrong - so if we have to follow SonarQube we would have to be on Java 15 or earlier (I.e. not cool).
2. Since SonarQube doesn't support Java 16 we're not any worse off if we move to 17.
3. I don't think SonarQube should get to dictate what language version we use. Sure, there are technologies that get to dictate: IDE, Gradle, Spring, etc, but no SonarQube. If this is a problem then we should find a replacement that can keep up.

### CODE REVIEW

**Review by kristianaasen — APPROVED** [review]

File: .github/workflows/gradle-branches.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Set up JDK 16
```
Comment (kristianaasen) [inline .github/workflows/gradle-branches.yml:15]: Set up JDK **17**

**Review by kristianaasen — APPROVED** [review]

File: .github/workflows/gradle-branches.yml (hunk 1 above)
Comment (pfs-olwe99) [reply .github/workflows/gradle-branches.yml:15]: Finns ingen egentlig anledning till att specificera version här. Stryk versionsnumret så slipper vi hålla det uppdaterat.

**Comment by pfs-olwe99** [comment]
Jag skulle helst se ett testprojekt som går före och testar nya versioner av kritiska komponenter som just Java innan vi inför det i en template.

**Comment by atomfinger** [comment]
> Jag skulle helst se ett testprojekt som går före och testar nya versioner av kritiska komponenter som just Java innan vi inför det i en template. 

@pfs-olwe99 kan jo oppgradere Bob integration å se hvordan det går? :D

**Review by pfs-olwe99 — APPROVED** [review]

**Pull request merged**